chore(pull_request): add pull request template (#958)
accuknox/KubeArmor: .github/pull_request_template.md

**Purpose of PR?**:

Fixes #

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

> **PLEASE REMOVE BELOW INFORMATION BEFORE SUBMITTING**

Reviewer: Just put it inside comments.
<!--
Your deets
-->

```md
<!--
Your deets
-->
```
![image](https://user-images.githubusercontent.com/47106543/199009547-0e2dd08b-9cca-482f-8661-711d5e7f4c91.png)

The PR title message must follow convention:
`<type>(<scope>): <subject>`.

Where: <br />
- `type` is to define what type of PR is this.
  Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script

>     - `chore`     - changes not related to production code

Reviewer 2: What's the production code?

Author: Production code mainly refers to the main branch on which changes are being done for the release.

Reviewer 2: All our code is done on `main` branch only and all the PR are pushed to it.

Author: Then production code refers to the main branch here. :)

    - `docs`      - changes related to documentation

>     - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes

Reviewer 2: Can we merge them as "docs" to simplify things?

Author: Style mostly means formatting or linting fix. Merging it with docs won't be right.

Reviewer 2: Yes I got that. Was trying to simplify things. No worries, we can keep it

    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

> - `scope` is a single word that best describes where the changes fit.

Reviewer 2: Does this mean the place where the changes are done (like system monitor)? It should be optional imho

Author: No, scope mostly means which feature, what kind of bugs or any related changes in ci. 
For ex: `feat(observability): <One line desc>`, `bug(policy..): ... `
Scope should be mandatory as it says for what features or bugs we are raising the PR.

Reviewer 2: It won't be easy for new contributors to determine it.

Author: That's why we have provided a checkbox, and we can always help new contributors and ask them to update the title. Although these things(features, bugs) has to be mentioned in the issue itself which can help new contributors to decide the title.

**Purpose of PR?**:

Fixes #

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

<!--

> **PLEASE REMOVE BELOW INFORMATION BEFORE SUBMITTING**

Reviewer: ```suggestion
```
no need for removing now :)

The PR title message must follow convention:
`<type>(<scope>): <subject>`.

Where: <br />
- `type` is to define what type of PR is this.
  Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script
    - `chore`     - changes not related to production code
    - `docs`      - changes related to documentation
    - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes
    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

> - `scope` is a single word that best describes where the changes fit.

Reviewer: should we add an example here? since the scope is limited to KubeArmor and not generic, we can mention
monitor,deployment,enforcer.....
Will help them determine.

> ## Pull Request template
> 
> Please, go through these steps before you submit a PR.
> 
> **Why is this PR required? What issue does it fix?**:
> 
> Fixes #
> 
> **What this PR does/why we need it?**:
> 
> Description
> 
> **Does this PR introduce a breaking change?** 
> 
> **If the changes in this PR are manually verified, list down the scenarios covered:**:
> 
> **Any additional information for your reviewer?** :
> _Mention if this PR is part of any design or a continuation of previous PRs_
> 
> 

Reviewer 3: ```suggestion
**Purpose of PR?**:

Fixes #

**Does this PR introduce a breaking change?** 

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

```
